test/cli/other_test.py

```diff
@@ -3164,6 +3164,61 @@ def test_check_unused_templates_func(tmp_path):  # #13714
     (void)(*((int*)0));
 }
 
+template<typename T>
+static inline void f_t_3()
+{
+    (void)(*((int*)0));
+}
```

Comment:
Needs to consider the test added in #7812 so the function names do not conflict. Maybe base this PR on the other one and include the commit for now.